File: generate-icons.js
// Simple icon generator using Canvas API (Node.js)
const fs = require('fs');
const { createCanvas } = require('canvas');

function generateIcon(size, outputPath) {
  const canvas = createCanvas(size, size);
  const ctx = canvas.getContext('2d');

  // Dark grey background
  ctx.fillStyle = '#1E1E1E';
  ctx.fillRect(0, 0, size, size);

  // White "S" letter
  ctx.fillStyle = '#FFFFFF';
  ctx.font = `bold ${size * 0.6}px Arial`;
  ctx.textAlign = 'center';
  ctx.textBaseline = 'middle';
  ctx.fillText('S', size / 2, size / 2);

  // Save to file
  const buffer = canvas.toBuffer('image/png');
  fs.writeFileSync(outputPath, buffer);
  console.log(`Generated ${outputPath}`);
}

const iconsDir = 'C:/Users/RUDRANKSH PARIAL/Documents/Project/SkelIO/icons';

generateIcon(16, `${iconsDir}/icon16.png`);
generateIcon(32, `${iconsDir}/icon32.png`);
generateIcon(48, `${iconsDir}/icon48.png`);
generateIcon(128, `${iconsDir}/icon128.png`);

console.log('All icons generated successfully!');

File: extension/generate_icons.py
from PIL import Image, ImageDraw, ImageFont
import os

# Create icons directory
icons_dir = r"C:\Users\RUDRANKSH PARIAL\Documents\Project\SkelIO\icons"
os.makedirs(icons_dir, exist_ok=True)

def create_icon(size, path):
    # Create image with dark grey background
    img = Image.new('RGB', (size, size), color='#1E1E1E')
    draw = ImageDraw.Draw(img)

    # Draw white "S" letter
    font_size = int(size * 0.6)
    try:
        # Try to use a system font
        font = ImageFont.truetype("arial.ttf", font_size)
    except:
        # Fallback to default font
        font = ImageFont.load_default()

    # Draw text centered
    text = "S"
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]

    x = (size - text_width) // 2
    y = (size - text_height) // 2

    draw.text((x, y), text, fill='#FFFFFF', font=font)

    # Save
    img.save(path)
    print(f"Created {path}")

# Generate all icon sizes
create_icon(16, os.path.join(icons_dir, "icon16.png"))
create_icon(32, os.path.join(icons_dir, "icon32.png"))
create_icon(48, os.path.join(icons_dir, "icon48.png"))
create_icon(128, os.path.join(icons_dir, "icon128.png"))

print("All icons generated successfully!")
